src/service/transaction/getTxList.ts

```diff
@@ -116,30 +117,36 @@ async function getTxTotalCount(data: GetTxListParam): Promise<number> {
   }
 
   if (data.action) {
-    distinctQuery = `${distinctQuery} AND type='${data.action}'`
+    distinctQuery = `${distinctQuery} AND type=$2`
+    params.push(data.action)
   }
 
   const totalCntQuery = `SELECT COUNT(*) FROM (${distinctQuery}) t`
-  const totalCntResult = await getConnection().query(totalCntQuery)
+  const totalCntResult = await getConnection().query(totalCntQuery, params)
   return +get(totalCntResult, '0.count', 0)
 }
 
 export async function getTxFromAccount(data: GetTxListParam, parse: boolean): Promise<GetTxsReturn> {
   if (!data.account) {
-    throw new Error(`Account address is required.`)
+    throw new TypeError(`Account address is required.`)
   }
 
-  const totalCnt = await getTxTotalCount(data)
+  if (!parseInt(data.limit as any, 10)) {
```

Comment:
do we actually need these? as page and limit already validated by Joi.

Reply:
Yes I am aware of that but to be more safe..